tests/test_transaction.py

```diff
@@ -199,12 +274,67 @@ def test_transaction_list(self):
+        return transaction_id, invoice_number, line_item_id
+ 
+    def _refund_transaction_by_line_items(self, transaction_id):
+        """
+        Given a transaction id, this method will refund all the line items on
+        the associated invoice.
+        """
+        transaction = recurly.Transaction.get(transaction_id)
+        invoice = transaction.invoice()
+        line_items = invoice.line_items
+        adjustments_to_refund = []
+        for line_item in line_items:
+            adjustments_to_refund.append({
+                'adjustment': line_item,
+                'quantity': line_item.quantity,
+                'prorate': False 
+            })
+        return invoice.refund(adjustments_to_refund)
+ 
+    def _assert_invoices_are_paired(self, original_invoice_number, type=FULL, partial_amount=None):
+        """
+        Given an ID that identifies the original invoice, this method asserts
+        that there is only a pair of invoices, where they relate to each other
+        in a refund manner.
+ 
+        type is full or partial. If partial, must also provided the partial amount.
+        """
+        self.assertEqual(len(mocurly.backend.invoices_backend.datastore), 2)
+        for invoice_number, invoice in mocurly.backend.invoices_backend.datastore.items():
+            self.assertEqual(len(invoice['transactions']), 1)
+            if invoice_number == original_invoice_number:
+                original_invoice = invoice
+            else:
+                refund_invoice = invoice
 
-        self.base_transaction_data['uuid'] = 'abcd'
-        mocurly.backend.transactions_backend.add_object('abcd', self.base_transaction_data)
+        self.assertEqual(len(mocurly.backend.adjustments_backend.datastore), 2)
+        for adjustment_uuid, adjustment in mocurly.backend.adjustments_backend.datastore.items():
+            if adjustment['invoice'] == original_invoice_number:
+                original_adjustment = adjustment
+            else:
+                refund_adjustment = adjustment
+        self.assertEqual(int(original_adjustment['quantity']), -int(refund_adjustment['quantity']))
 
-        acc = recurly.Account.get(self.base_account_data['uuid'])
-        transactions = list(acc.transactions())
-        self.assertEqual(len(transactions), 2)
-        self.assertEqual(set([transaction.uuid for transaction in transactions]), set(['1234', 'abcd']))
+        if type == FULL:
+            self.assertEqual(original_invoice['total_in_cents'], -refund_invoice['total_in_cents'])
+            self.assertEqual(original_adjustment['total_in_cents'], -refund_adjustment['total_in_cents'])
+        else:
+            self.assertEqual(refund_invoice['total_in_cents'], -partial_amount)
+            self.assertEqual(refund_adjustment['total_in_cents'], -partial_amount)
+        return original_invoice, refund_invoice, original_adjustment, refund_adjustment
```

Comment:
If you look at the function in isolation, it makes perfect sense from a design perspective why this should return `original_adjustment` and `refund_adjustment` (it parallels the invoices).

While I can remove it as dead code, then it begs the question why it doesn't return it when you look at this function in isolation, in which case I have to write documentation explaining that the current tests do not need to do additional checks on the adjustment objects. OR I can refactor this so that it is two stages.

Anyway, that was my original logic. I don't feel strongly, so if you feel really strongly about removing dead code, I will remove.